test/gpb_compile_tests.erl

```diff
@@ -1665,6 +1666,31 @@ nif_with_strbin() ->
                 end)
       end).
 
+nif_with_booleans() ->
+    with_tmpdir(
+      fun(TmpDir) ->
+              M = gpb_nif_with_booleans,
+              DefsTxt = lf_lines(["message ntest3 {",
+                                  "    required bool b = 1;",
+                                  "}"]),
+              Defs = parse_to_proto_defs(DefsTxt),
+              {ok, Code} = compile_nif_msg_defs(M, DefsTxt, TmpDir, []),
+              in_separate_vm(
+                TmpDir, M, Code,
+                fun() ->
+                        OrigMsgAtom = {ntest3,true},
+                        OrigMsgInt = {ntest3,1},
+                        MEncoded  = M:encode_msg(OrigMsgAtom),
+                        GEncoded  = gpb:encode_msg(OrigMsgAtom, Defs),
+                        MMDecoded = M:decode_msg(MEncoded, ntest3),
+                        GMDecoded = gpb:decode_msg(MEncoded, ntest3, Defs),
+                        MGDecoded = M:decode_msg(GEncoded, ntest3),
+                        ?assertEqual(OrigMsgAtom, MMDecoded),
+                        ?assertEqual(OrigMsgInt, GMDecoded),
```

Comment:
I think this line should read: `?assertEqual(OrigMsgAtom, GMDecoded),`
(Ie `OrigMsgAtom` instead of `OrigMsgInt`, making the `OrigMsgInt` variable unused)

I found this out during a test merge I did of all `epb` compat branches, it probably went unnoticed earlier due to the other bug of not catching bugs in from the separate vm stage, sorry about that :/